src/functions/sendCandidateResults/application/service/send-notification.ts

```diff
@@ -0,0 +1,109 @@
+import { DocumentsServiceError } from '../../domain/errors/documents-service-error';
+import { PersonalisationDetails } from '../../domain/personalisation.model';
+import { INotifyClient } from '../../domain/notify-client.interface';
+import { get } from 'lodash';
+import { AxiosError } from 'axios';
+import * as Handlebars from 'handlebars';
+import { subjectMapper, templateMapper } from './template-selector';
+import { Language } from '../../domain/conducted-language';
+import { TestOutcome } from '../../domain/test-outcome';
+import { CommunicationMethod } from '@dvsa/mes-test-schema/categories/common';
+import { Correspondence } from '../../domain/template-id.model';
+
+export async function sendNotification(
+  emailAddress: string,
+  communicationMethod: CommunicationMethod,
+  templateId: string,
+  notificationPersonalisation: PersonalisationDetails,
+  reference: string,
+  emailReplyToId: string,
+  client: INotifyClient,
+  conductedLanguage: Language,
+  testOutcome: TestOutcome,
+  padi?: boolean,
+): Promise<any> {
+
+  try {
+    let address;
+    let renderedSubject;
+
+    let personalisation = {};
+
+    try {
+      const compileSubject = Handlebars.compile(
+        subjectMapper(notificationPersonalisation.category, conductedLanguage, padi)
+      );
+
+      renderedSubject = compileSubject(notificationPersonalisation);
+
+      personalisation = {
+        ...personalisation,
+        renderedSubject,
+      };
+    } catch (error) {
+      console.error('Error preparing renderedSubject', error);
+      return;
```

Comment:
Are we better off throwing the error here and handling that in the calling method?

Reply:
cos ive got a try/catch within a try/catch i cant throw the error cos then the return will be prevented from stopping the execution.  will investigate

Reply:
reworked to not have a try catch within a try catch...each will progress seperatly and throw apporapreatly.